### DISCUSSION TITLE
skynet1.5版本M1芯片下，require一个c库在gc处crash

### DISCUSSION THREAD

**Discussion by lailongwei** [cloudwu/skynet#1757] [General]
如题，require一个c库，引发crash，是在lua_newtable代码的checkGC调用处，crash时的l->gcstate及其他变量已经全部野了，怀疑是lua state已经被释放，想问下大家有遇到这个问题吗？
PS：lua以库没有任何特殊代码，就一个lua_newlib调用，同时在非skynet环境require就没有问题
@cloudwu 大大可以帮忙看一下吗？

**Comment by cloudfreexiao** [comment]
先用最新代码测试下看看

**Comment by cloudwu** [comment]
检查 lua 是否有正确链接，避免重复链接。
